### PR TITLE
test(plugin-phone): expand test coverage

### PR DESCRIPTION
**Pull request by adamweeks**
Ready for review. Will be much cleaner once #429 is merged in.

Still quite a few tests skipped or incomplete, but this is a lot more coverage for the `plugin-phone` as it exists now.

### CODE REVIEW

File: packages/plugin-phone/test/integration/spec/call.js [hunk 1]
```diff
@@ -223,47 +320,161 @@ describe(`plugin-phone`, function() {
...
-        it(`is false`);
+        it(`is false`, () => {
+          return call.stopSendingAudio()
+            .then(() => assert.isFalse(call.sendingAudio));
+        });
       });
     });
 
     describe(`#sendingVideo`, () => {
+      let call;
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/call.js]: Does it make sense to do these tests based on starting the calls with different constraints rather instead of changing the audio status in the call?

File: packages/plugin-phone/test/integration/spec/call.js [hunk 2]
```diff
@@ -223,47 +320,161 @@ describe(`plugin-phone`, function() {
...
+      afterEach(() => {
+        const c = call;
+        call = undefined;
+        return c.hangup().catch((reason) => console.warn(reason));
+      });
+
       describe(`when the local party is receiving Video`, () => {
-        it(`is true`);
+        it(`is true`, () => {
+          return call.startReceivingVideo()
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/call.js]: use implicit returns when possible

File: packages/plugin-phone/test/integration/spec/call.js [hunk 3]
```diff
@@ -317,18 +576,135 @@ describe(`plugin-phone`, function() {
...
+
+      describe(`in-progress events`, () => {
+        let call;
+        beforeEach(() => {
+          call = spock.spark.phone.dial(mccoy.email);
+          assert.equal(call.status, `initiated`);
+        });
+
+        describe(`on(remoteAudioMuted:change)`, () => {
+          // stopSendingAudio doesn't trigger this change but should
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/call.js]: can you change the comment to something more like "when the remote part invokes stop sending audio, the local party doesn't receive the update"? I had to stare at it for a while before I realized the comment was, in fact, valid.

File: packages/plugin-phone/test/integration/spec/call.js [hunk 4]
```diff
@@ -317,18 +576,135 @@ describe(`plugin-phone`, function() {
...
+                .then(() => remoteCall.stopSendingAudio())
+                .then(() => remoteCall.hangup()),
+              call.when(`disconnected`)
+                .then(() => {
+                  return assert.callCount(remoteAudioSpy, 1);
+                })
+            ]);
+          });
+
+          // startSendingAudio doesn't trigger this change but should
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/call.js]: see previous

File: packages/plugin-phone/src/call.js [hunk 5]
```diff
@@ -389,7 +389,9 @@ const Call = SparkPlugin.extend({
       this.stopListening(this.spark.mercury);
       this.off();
       URL.revokeObjectURL(this.localMediaStreamUrl);
+      this.localMediaStreamUrl = undefined;
```
Comment (adamweeks) [inline packages/plugin-phone/src/call.js]: This is the only "fix" I put in for the plugin to match the tests. We will do more in another PR.

**Comment by adamweeks** [comment]
Reasons for `skip` tests:
- If a call is created and `.hangup()` is called on it before the locus session is created, the promise just waits for the locus session until timeout. We need a way to cancel a locus creation.
- `remoteAudioMuted:change` does not get called when the remote caller changes their audio status (with `.stopSendingAudio()` )
- When calling `.toggleReceivingAudio()`  to change into send only status, the sdp media direction should be `sendonly` but it is actually `inactive` 
- Test users cannot create 1:1 messages to each other (some sort of message signing issue perhaps?).

File: packages/plugin-phone/test/integration/spec/phone.js [hunk 6]
```diff
@@ -65,19 +64,38 @@ describe(`plugin-phone`, function() {
     });
 
     describe(`#deregister()`, () => {
-      it(`disconnects from mercury`);
-      it(`unregisters from wdm`);
-      it(`is a noop when not registered`);
+      it(`disconnects from mercury`, () => {
+        const mercuryDisconnectSpy = sinon.spy(spock.spark.mercury, `disconnect`);
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/phone.js]: add assertions that mercury is connected to demonstrate the test is valid

File: packages/plugin-phone/test/integration/spec/phone.js [hunk 7]
```diff
@@ -65,19 +64,38 @@ describe(`plugin-phone`, function() {
...
+        return spock.spark.phone.deregister()
+          .then(() => assert.calledOnce(mercuryDisconnectSpy))
+          .then(() => assert.isFalse(spock.spark.mercury.connected, `Mercury is not connected`))
+          .then(() => assert.isFalse(spock.spark.phone.connected, `Mercury (proxied through spark.phone) is not conneted`))
+          .then(() => mercuryDisconnectSpy.restore());
+      });
+
+      it(`unregisters from wdm`, () =>
+        spock.spark.phone.deregister()
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/phone.js]: implicit returns should go on the same line as the `=>`

File: packages/plugin-phone/test/integration/spec/phone.js [hunk 8]
```diff
@@ -65,19 +64,38 @@ describe(`plugin-phone`, function() {
...
+          .then(() =>
+            spock.spark.device.on(`change:device`)
+              .then(() => assert.isUndefined(spock.spark.device.url))
+          )
+      );
+
+      // TODO: Does not currently noop after multiple calls
+      it.skip(`is a noop when not registered`, () => {
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/phone.js]: mercury.disconnect() is the noop in question. maybe spy on spark.request (which is currently impossible, iirc) instead and assert that no network calls get made?

File: packages/plugin-phone/test/integration/spec/phone.js [hunk 9]
```diff
@@ -138,7 +154,35 @@ describe(`plugin-phone`, function() {
 
       it(`calls a user by AppID username`);
       it(`calls a user by tropo uri`);
-      it(`calls a user by spark uri`);
+
+      // FIXME: Test users currently cannot create 1:1 messages to each other
+      it.skip(`calls a user by spark uri`, () => {
+        let conversation;
+        let sparkUri;
+        return spock.spark.request({
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/phone.js]: can you write a testcase somewhere demonstrating the 1:1 messaging problem?

File: packages/plugin-phone/test/integration/spec/media-state-controls.js [hunk 10]
```diff
@@ -144,8 +145,9 @@ describe(`plugin-phone`, function() {
         });
 
         describe(`when the call is not receiving audio`, () => {
```
Comment (ianwremmel) [inline packages/plugin-phone/test/integration/spec/media-state-controls.js]: Before eliminating the bugzilla comment, please run this test in firefox via sauce. I think you'll find that it fails with a different error

**Comment by ianwremmel** [comment]
@adamweeks  this is merged, right?

**Comment by ianwremmel** [comment]
closing in favor of alternate merge method
